### DISCUSSION TITLE
How to only allow API requests from Next.js app

### DISCUSSION THREAD

**Discussion by koji98** [vercel/next.js#31619] [Help] [answered]
Does anyone know how to secure the serverless API routes to only be accessible from the web application itself? The app does not require user auth. I want to make sure other websites and servers can not make requests to the API. Only users on the site should be able to take actions that request the API routes.

**Comment by jamesmosier** [comment, marked as answer]
By default, API routes are [same-origin only](https://nextjs.org/docs/api-routes/introduction#caveats). Therefore they are secured to your application by default.

**Reply by koji98** [reply]
I came across [this](https://github.com/vercel/next.js/discussions/16441) discussion and believed that the API is still accessible by different servers. In that discussion thread, they said they're able to postman their deployed app. I have not deployed yet but wanted to make sure to not allow this.

**Reply by jamesmosier** [reply]
Correct, this only protects against other web applications from calling your API per [this comment](https://github.com/vercel/next.js/issues/17814#issuecomment-707029253).

If you want to ensure no one else is calling your API, you'll need to implement some sort of authentication to verify the sender of the request. 

One solution might be to use something like [next-auth](https://next-auth.js.org/tutorials/securing-pages-and-api-routes).

**Reply by koji98** [reply]
I will check it out, thank you!

**Reply by tilxyz** [reply]
I don't the  [comment](https://github.com/vercel/next.js/issues/17814#issuecomment-707029253) is correct. I am able to make an API from a chrome extension in the browser, the origin is `chrome-extension://<chrome_extension_id>`, which is obviously different from my domain, but it still passes.

**Reply by tilxyz** [reply]
Do we know why it's possible?
